modules/ROOT/pages/index.adoc

```diff
@@ -1,77 +1,82 @@
 [[index]]
 = Introduction
 :page-aliases: introduction.adoc
-:description: This section describes the Neo4j GraphQL Library.
+:description: An introduction to the Neo4j GraphQL Library.
 
-The Neo4j GraphQL Library is a highly flexible, low-code, open source JavaScript library that enables rapid API development for cross-platform and mobile applications by tapping into the power of connected data.
+The Neo4j GraphQL Library is:
 
-With Neo4j as the graph database, the GraphQL Library makes it simple for applications to have data treated as a graph natively from the frontend all the way to storage.
-This avoids duplicate schema work and ensures flawless integration between frontend and backend developers.
-
-If you are new to Neo4j and GraphQL take a look at xref:getting-started/index.adoc[Creating a new project] and xref:getting-started/toolbox.adoc[Neo4j GraphQL Toolbox] to learn the fundamentals of the Neo4j GraphQL Library and how to create GraphQL APIs backed by a Neo4j graph database.
+* highly flexible and low-code
+* open source
+* a JavaScript library
+* leveraging graph-native data
 
+It enables rapid API development for cross-platform and mobile applications.
 
 == How it works
 
-The Neo4j GraphQL Library requires a set of type definitions that describes the shape of your graph data.
-It can generate an entire executable schema with all of the additional types needed to execute queries and mutations to interact with your Neo4j database.
+The Neo4j GraphQL Library requires a set of type definitions that describes the shape of your graph data and creates an API layer to communicate with the data.
+
+It generates an entire executable schema with all additional types needed to execute queries and mutations to interact with your Neo4j database.
+
+For example, consider these type definitions:
+
+[source, graphql, indent=0]
+----
+type Product @node {
+    productName: String
+    category: [Category!]! @relationship(type: "PART_OF", direction: OUT)
+}
+
+type Category @node {
+    categoryName: String
+    products: [Product!]! @relationship(type: "PART_OF", direction: IN)
+}
+----
+
+Based on these type definitions, the library generates query and mutation templates to create new instances of the types as well as query existing instances.
+
+The following mutation creates a new product as well as a new category:
+
+[source, graphql, indent=0]
+----
+mutation {
+  createProducts(
+    input: [
+      {
+        productName: "New Product"
+        category: { create: [{ node: { categoryName: "New Category" } }] }
+      }
+    ]
+  ) {
+    products {
+      productName
+      category {
+        categoryName
+      }
+    }
+  }
+}
+----
+
+Here is an example of how you can query existing data:
+
+[source, graphql, indent=0]
+----
+query {
+  products {
+    productName
+    category {
+      categoryName
+    }
+  }
+}
+----
```

Comment:
I would consider putting the query first, along with an example response, as this feels like the most prominent and use case